protos/core.proto

```diff
@@ -247,6 +247,190 @@ message NotebookRuntimeOptions {
   }
 }
 
+// Data Preparation Related entries
+message DataPreparation {
+  // Data preparatiohs can have more than 1 output
+  repeated Target targets = 1;
+
+  repeated Target canonical_targets = 2;
+
+  repeated string tags = 3;
+
+  repeated Target dependency_targets = 4;
+
+  string file_name = 5;
+
+  bool disabled = 6;
+
+  DataPreparationDefinition data_preparation = 7;
+}
+
+message DataPreparationDefinition {
+  repeated DataPreparationNode nodes = 1;
+  DataPreparationGenerated generated = 2;
+}
```

Comment:
These values need to be populated from somewhere, I'd imagine that the config proto needs more options that roughly correspond to the data in here?

Reply:
Yes, the Data Preparation YAML file is formatted according to this proto. During processing, we will parse the data preparation YAML and build this proto object from the parse result.